merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java

```diff
@@ -713,16 +717,24 @@ public InProgress continueWith(final Task newState) {
     }
 
     /** The task has completed its primary operation, but has unfinished children. */
-    record AwaitingChildren(Duration startOffset, Duration endOffset, LinkedList<TaskId> remainingChildren)
-        implements ExecutionState
+    record AwaitingChildren(
+        Duration startOffset,
+        Duration endOffset,
+        LinkedList<TaskId> remainingChildren,
+        TaskReturnValue<?> returnValue
+    ) implements ExecutionState
     {
-      public Terminated joinedAt(final Duration joinOffset) {
-        return new Terminated(this.startOffset, this.endOffset, joinOffset);
+      public Terminated joinedAt(final Duration joinOffset, final TaskReturnValue<?> returnValue) {
+        return new Terminated(this.startOffset, this.endOffset, joinOffset, returnValue);
```

Comment:
`AwaitingChildren` already has a `returnValue` field it can forward to the `Terminated` variant. Since `joinedAt` accepts a whole other `returnValue`, this field is (locally) unused. I imagine calling code extracts it out and then passes it back in, but this is unnecessary.